src/client/storage/uploader.ts

```diff
@@ -90,3 +102,21 @@ export const uploadLoop = async () => {
 
     setTimeout(uploadLoop, UPLOAD_LOOP_INTERVAL);
 };
+
+export const chunkValidatorLoop = async () => {
+
+    const allCompletedChunks: any[] = await Chunk.allBy('ul_status', CHUNK_UPLOAD_STATUS.VALIDATING, false);
+    allCompletedChunks.forEach(async (chunk) => {
+
+        try {
+            await arweave.transactions.getData(chunk.txid, {decode: true});
+            chunk.ul_status = CHUNK_UPLOAD_STATUS.COMPLETED;
+        } catch (error) {
```

Comment:
Why are you setting it to failed after the first failed attempt? We should implement some timeout or limit, after which it should be treated as failed. And, I think, this limit should be the DB field, same as we have for upload loop